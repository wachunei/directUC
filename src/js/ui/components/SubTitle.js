import styled from "styled-components";
import Title from "./Title";

const SubTitle = styled(Title).attrs({ as: "h3" })`
  font-size: 18px;
`;

export default SubTitle;
