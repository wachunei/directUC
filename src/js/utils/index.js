export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
