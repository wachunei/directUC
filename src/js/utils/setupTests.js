// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
