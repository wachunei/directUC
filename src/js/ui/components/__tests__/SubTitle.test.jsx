import SubTitle from "../SubTitle";
import { render, getElement } from "./utils";

describe("SubTitle Component", () => {
  test("should match snapshot", () => {
    const { container } = render(<SubTitle />);
    expect(getElement(container)).toMatchSnapshot();
  });
  test("should match snapshot with lightTheme", () => {
    const { container } = render(<SubTitle />, "lightTheme");
    expect(getElement(container)).toMatchSnapshot();
  });
  test("should match snapshot with darkTheme", () => {
    const { container } = render(<SubTitle />, "darkTheme");
    expect(getElement(container)).toMatchSnapshot();
  });
});
