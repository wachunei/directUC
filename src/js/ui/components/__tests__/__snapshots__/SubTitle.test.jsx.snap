// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle Component should match snapshot 1`] = `
.c0 {
  display: block;
  font-size: 24px;
  font-weight: 500;
  color: #303030;
  font-family: "Red Hat Display",sans-serif;
  margin: 0 0 0.5rem;
}

.c1 {
  font-size: 18px;
}

<h3
  class="c0 c1"
/>
`;

exports[`SubTitle Component should match snapshot with darkTheme 1`] = `
.c0 {
  display: block;
  font-size: 24px;
  font-weight: 500;
  color: #E0E0E0;
  font-family: "Red Hat Display",sans-serif;
  margin: 0 0 0.5rem;
}

.c1 {
  font-size: 18px;
}

<h3
  class="c0 c1"
/>
`;

exports[`SubTitle Component should match snapshot with lightTheme 1`] = `
.c0 {
  display: block;
  font-size: 24px;
  font-weight: 500;
  color: #303030;
  font-family: "Red Hat Display",sans-serif;
  margin: 0 0 0.5rem;
}

.c1 {
  font-size: 18px;
}

<h3
  class="c0 c1"
/>
`;
